File: DeskHubSharpRevised/Models/RequestList.cs
using System.Collections.ObjectModel;

namespace DeskHubSharpRevised.Models;

public class RequestList
{
    /// <summary>
    /// Stores everything in User, typically from the request
    /// </summary>
    public static User userDetail { get; set; }

    /// <summary>
    /// Stores everything sent in Branch, typically from request
    /// </summary>
    public static ObservableCollection<Branch> branchDetail { get; set; }

    /// <summary>
    /// Stores everything sent in RepoDetail, typically from request
    /// </summary>
    public static ObservableCollection<RepoDetail> repoDetail { get; set; }
}

File: ModelsLib/RequestList.fs
namespace DeskHubSharpRevised.Models

open System.Collections.ObjectModel

type RequestList() =
    static member val userDetail = new User()
        with get, set
    static member val branchDetail = new ObservableCollection<Branch>()
        with get, set
    static member val repoDetail = new ObservableCollection<RepoDetail>()
        with get, set